FTBFS php 8.5.6 with GCC 16

### Description

Hi,
when building php 8.5.6 with GCC 16, I get the following error:
```
[  220s] In file included from /home/abuild/rpmbuild/BUILD/php8-8.5.6-build/php-8.5.6/ext/opcache/jit/zend_jit.c:43:
[  220s] /home/abuild/rpmbuild/BUILD/php8-8.5.6-build/php-8.5.6/ext/opcache/jit/zend_jit.c: In function ‘zend_runtime_jit’:
[  220s] ext/opcache/jit/zend_jit_internal.h:215:70: error: cannot tail-call: caller uses setjmp
[  220s]   215 | # define ZEND_OPCODE_RETURN()                  ZEND_OPCODE_TAIL_CALL((zend_vm_opcode_handler_t)opline->handler)
[  220s]       |                                                                      ^
[  220s] ext/opcache/jit/zend_jit_internal.h:213:39: note: in definition of macro ‘ZEND_OPCODE_TAIL_CALL’
[  220s]   213 |                 ZEND_MUSTTAIL return (handler)(ZEND_OPCODE_HANDLER_ARGS_PASSTHRU); \
[  220s]       |                                       ^~~~~~~
[  220s] /home/abuild/rpmbuild/BUILD/php8-8.5.6-build/php-8.5.6/ext/opcache/jit/zend_jit.c:3140:9: note: in expansion of macro ‘ZEND_OPCODE_RETURN’
[  220s]  3140 |         ZEND_OPCODE_RETURN();
[  220s]       |         ^~~~~~~~~~~~~~~~~~
[  220s] make: *** [Makefile:2301: ext/opcache/jit/zend_jit.lo] Error 1
[  220s] make: *** Waiting for unfinished jobs....
```

I am aware of issue #21960, but the fix mentioned there doesn't work for me (the error above is with LTO disabled). Full build log attached.

[build_nolto.log](https://github.com/user-attachments/files/28499565/build_nolto.log)

### PHP Version

```plain
PHP 8.5.6
```

### Operating System

openSUSE Tumbleweed

--- a/Zend/Zend.m4
+++ b/Zend/Zend.m4
@@ -317,6 +317,7 @@ int emu(const opcode_handler_t *ip, void *fp) {
   while ((*ip)());
   FP = orig_fp;
   IP = orig_ip;
+  return 0;
 }], [])],
   [php_cv_have_global_register_vars=yes],
   [php_cv_have_global_register_vars=no])

